cmake/HalideTestHelpers.cmake

```diff
@@ -26,12 +26,14 @@ if (NOT TARGET Halide::Test)
 
     # Tests are built with the equivalent of OPTIMIZE_FOR_BUILD_TIME (-O0 or /Od).
     # Also allow tests, via conditional compilation, to use the entire
-    # capability of the CPU being compiled on via -march=native. This
-    # presumes tests are run on the same machine they are compiled on.
+    # capability of the CPU being compiled on via -march=native if the
+    # the compiler supports it. This also presumes tests are run on the
+    # same machine they are compiled on.
     target_compile_options(Halide_test INTERFACE
                            $<$<CXX_COMPILER_ID:MSVC>:/Od>
                            $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-O0>
-                           $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-march=native>)
+                           $<$<CXX_COMPILER_ID:Intel>:-march=native>
+                           $<$<CXX_COMPILER_ID:GNU>:-march=native>)
```

Comment:
tbqh we should just remove this because it's going to make cross compiling Halide harder.